hs-abci-sdk/src/Tendermint/SDK/Modules/Validators/Keeper.hs

```diff
@@ -0,0 +1,94 @@
+{-# LANGUAGE TemplateHaskell #-}
+
+module Tendermint.SDK.Modules.Validators.Keeper where
+
+import qualified Data.Map.Strict                         as Map
+import           Data.Maybe                              (fromMaybe)
+import qualified Data.Set                                as Set
+import           Data.Word                               (Word64)
+import           Polysemy                                (Members, Sem,
+                                                          interpret, makeSem)
+import           Polysemy.Error                          (Error)
+import           Tendermint.SDK.BaseApp                  (AppError, ReadStore,
+                                                          WriteStore)
+import qualified Tendermint.SDK.BaseApp.Store.List       as L
+import qualified Tendermint.SDK.BaseApp.Store.Map        as M
+import qualified Tendermint.SDK.BaseApp.Store.Var        as V
+import           Tendermint.SDK.Modules.Validators.Store
+import           Tendermint.SDK.Modules.Validators.Types
+
+
+data ValidatorsKeeper m a where
+  GetValidators :: ValidatorsKeeper m (Map.Map PubKey_ Word64)
+  GetValidatorsKeys :: ValidatorsKeeper m (Set.Set PubKey_)
+  GetPowerOf :: PubKey_ -> ValidatorsKeeper m Word64
+  GetQueuedUpdates :: ValidatorsKeeper m (Map.Map PubKey_ Word64)
+  GetQueuedUpdate :: PubKey_ -> ValidatorsKeeper m (Maybe Word64)
+  QueueUpdate :: PubKey_ -> Word64 -> ValidatorsKeeper m ()
+  QueueUpdates :: [(PubKey_, Word64)] -> ValidatorsKeeper m ()
```

Comment:
You don't actually need `QueueUpdates` as a summand since it's implemented in terms of folding `QueueUpdate`, you can just write a function 

`queueUpdates :: Members ValidatorsEffs r => [(PubKey_, Word64)] -> Sem r ()` 

directly.

Reply:
Fair point. Will remove.